enzyme/Enzyme/MLIR/Passes/RemovalUtils.cpp

```diff
@@ -0,0 +1,40 @@
+//===- RemovalUtils.cpp - Utilities to remove Enzyme ops -------* C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "RemovalUtils.h"
+#include "Interfaces/AutoDiffOpInterface.h"
+
+mlir::Type mlir::enzyme::CacheInfo::batchType() {
+  return mlir::enzyme::CacheInfo::batchType(mlir::ShapedType::kDynamic);
+}
+
+mlir::Type mlir::enzyme::CacheInfo::batchType(int64_t dim) {
```

Comment:
so there is already an Enzyme Autodiff Type interface, which should have a method for batching (and if not that would probably be the right place for this)